src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -151,6 +151,58 @@ void CascadiaSettings::_CreateDefaultProfiles()
     _profiles.emplace_back(powershellProfile);
     _profiles.emplace_back(cmdProfile);
     
+    // Add all WSL distros from: HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+    Profile WSLDistro{};
+    wil::unique_hkey hLXSSKey;
+    wil::unique_hkey hHKCUUncached;
+    RegOpenCurrentUser(KEY_READ, &hHKCUUncached); // Retrieve the uncached equivalent of HKEY_CURRENT_USER

```

Comment:
My reading of the msdn page for this function suggests that this is actually the *cached* result?

Reply:
> The HKEY_CURRENT_USER key maps to the root of the current user's branch in the HKEY_USERS key. It is cached for all threads in a process. Therefore, this value does not change when another user's profile is loaded.

The way I'm reading it is the `HKEY_CURRENT_USER `macro will use the path under `HKEY_USERS`.

> RegOpenCurrentUser uses the thread's token to access the appropriate key, or the default if the profile is not loaded.

I read this portion as saying that the `RegOpenCurrentUser `function has access to the user token allowing it to access the true/active `HKEY_CURRENT_USER`.

Reply:
https://docs.microsoft.com/en-us/windows/desktop/api/winreg/nf-winreg-regopenkeyexw
> If your service or application **impersonates different users**, do not use this function with HKEY_CURRENT_USER. Instead, call the RegOpenCurrentUser function.

I think RegOpenCurrentUser is for such condition, here we can use RegOpenKeyExW with HKEY_CURRENT_USER.